src/game/client/components/chat.cpp

```diff
@@ -430,7 +430,7 @@ void CChat::AddLine(int ClientID, int Mode, const char *pLine)
 			{
 				int Length = str_length(m_pClient->m_aClients[m_pClient->m_LocalClientID].m_aName);
 				if((pLine == pHL || pHL[-1] == ' ') // space or nothing before
-					&& (((pHL[Length] == 0 || pHL[Length] == ' ') || pHL[Length] == ':') && (pHL[Length+1] == 0 || pHL[Length+1] == ' '))) // space or nothing after, allowing a colon
+					&& ((pHL[Length] == 0 || pHL[Length] == ' ') || (pHL[Length] == ':' && (pHL[Length+1] == 0) || pHL[Length+1] == ' '))) // space or nothing after, allowing a colon
```

Comment:
Should be
`&& ((pHL[Length] == 0 || pHL[Length] == ' ') || (pHL[Length] == ':' && (pHL[Length+1] == 0 || pHL[Length+1] == ' ')))) // space or nothing after, allowing a colon` 
right? Otherwise pHL[Length] could be anything as long as pHL[Length+1] is ' '.

Reply:
Dammit, you are right. Makes me think this kind of complex logical expressions should be split in several `if`.

Reply:
Yeah, probably better.